state/remoteapplication.go

```diff
@@ -50,6 +53,29 @@ type remoteEndpointDoc struct {
 	Scope     charm.RelationScope `bson:"scope"`
 }
 
+type attributeMap map[string]interface{}
+
+// RemoteSpace represents a space in another model that endpoints are
+// bound to.
+type RemoteSpace struct {
+	CloudType          string         `bson:"cloud-type"`
+	Name               string         `bson:"name"`
+	ProviderId         string         `bson:"provider-id"`
+	ProviderAttributes attributeMap   `bson:"provider-attributes"`
+	Subnets            []RemoteSubnet `bson:"subnets"`
+}
+
+// RemoteSubnet represents a subnet in another model. Unfortunately we
+// can't reuse state.Subnet - it's tied to the subnets collection.
```

Comment:
We don't need this last sentence. We would not want to re-use state.Subnet regardless as that would be a layer violation.